### PR TITLE
Convert to CMake

### PR DESCRIPTION
**Pull request by fntlnz**
Converting the repo to CMake, usage in the readme.

This is a work in progress, opened the PR to give everyone a chance to review early.

Todo:

- [x] convert repo to cmake
- [x] make it work on jenkins
- [x] uncomment aarch64 on jenkins while building in a different directory for each architecture
- [x] make sure all targets rebuild if sources are changed

### CODE REVIEW

File: GPL/HostIsolation/TcFilter/Makefile [hunk 1]
```diff
@@ -37,7 +37,7 @@ TEST_FILES ?= $(wildcard *Tests.cpp)
 
 # We never want the stdinc flags in our BPF program, they can contain
 # things the eBPF program cannot compile or provide LLVM builtins that we can't use.
-NOSTDINC_FLAGS += -nostdinc -isystem $(shell $(CC) -print-file-name=include)
+NOSTDINC_FLAGS += -nostdinc -isystem $(shell $(CLANG) -print-file-name=include)
```
Comment (stanek-michal) [inline GPL/HostIsolation/TcFilter/Makefile:40]: Do we still need this Makefile? It seems `GPL/HostIsolation/TcFilter/CMakeLists.txt` covers it

File: non-GPL/HostIsolationMapsUtil/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# Demo using the library
+add_executable(UpdateIPsDemo UpdateIPsDemo.c)
+target_include_directories(UpdateIPsDemo PRIVATE 
+    "${LIBBPF_TARGET_INCLUDE_DIR}"
+    "${LIBBPF_UAPI_INCLUDE_DIR}"
+    "${LIBBPF_INCLUDE_DIR}" 
+    "${PROJECT_SOURCE_DIR}/non-GPL/Common")
+
+target_link_libraries(UpdateIPsDemo UpdateMaps)
```
Comment (stanek-michal) [inline non-GPL/HostIsolationMapsUtil/CMakeLists.txt:28]: looks like UpdatePidsDemo is missing

File: README.md [hunk 3]
```diff
@@ -1,82 +1,102 @@
...
+mkdir build
+cd build
+cmake ..
+make
 ```
 
-Or if you want to use a custom path for the eBPF object file.
+Besides the usual CMake variables, you can set the following variables which are specific to this project.
+
+| Variable      | Description                                      |
```
Comment (stanek-michal) [inline README.md:77]: needs another newline

**Review by stanek-michal — APPROVED** [review]
Looks like the old TcFilter Makefile can be removed. Also, UpdatePidsDemo needs an entry in CMakeLists.txt. 
Otherwise LGTM, great work! The build system looks a lot more sane now.

**Comment by fntlnz** [comment]
Thanks for the findings @stanek-michal  ❤️ 

I see many approvals here now, thanks for the time everyone.

I'm working on making Jenkins work here. I'll merge once I have everything working in there to make extra sure we don't break anything

File: non-GPL/HostIsolationMapsUtil/CMakeLists.txt (hunk 2 above)
Comment (fntlnz) [reply non-GPL/HostIsolationMapsUtil/CMakeLists.txt:28]: ah good catch, I forgot about that one! ty!

File: GPL/HostIsolation/TcFilter/Makefile (hunk 1 above)
Comment (fntlnz) [reply GPL/HostIsolation/TcFilter/Makefile]: removed this and addressed your other suggestions. Added a co-authored by to the commit. Thanks!

File: Jenkinsfile [hunk 4]
```diff
@@ -46,26 +47,24 @@ def generateTestClosure(arch, machine_name)
         node(machine_name)
         {
             // Everything within node() gets run on the target machine
-
+            sh 'uname -r'
```
Comment (mjwolf) [inline Jenkinsfile:50]: I guess this isn't meant to be left in, but it still would be good info to always have. Maybe you could move it into the println below

**Pull request merged**